### PR TITLE
new open weather fact gathering module

### PR DESCRIPTION
**Pull request by bcoca**

### CODE REVIEW

**Comment by gregdek** [comment]
LOL -- so who wants to review @bcoca's P5 module for openweather facts?

I nominate @resmo just because.

**Comment by bcoca** [comment]
this was actually part of my tips n tricks talk, but left it out as I thought it already had too much info

**Comment by resmo** [comment]
@gregdek I even got a use case for this :)

**Comment by bcoca** [comment]
I had one, but it involved a 10yr contract w/o SLA and a datacenter with marginal AC and machines w/o usable on board temp sensors .... fun days!

**Comment by gregdek** [comment]
Updating with the new process. We will be evaluating all new module PRs according to this process, effective immediately.

# 

Thanks for submitting this new module to Ansible Extras! This module is now in community review, a process that is open to all Ansible users. In order for this module to be approved, it must gain the following votes:

“works_for_me”: If you have tested the module thoroughly, including testing of all of the module’s options, and if the module works for you, please add “works_for_me” in the comments.

“passes_guidelines”: If you have gone through the module guidelines and the module meets all of the requirements, please add “passes_guidelines” in the comments.

“needs_revision”: If the module fails to work for you, or if it doesn’t meet guidelines, please add “needs_revision” in the comments with details about what needs to be fixed.

When a module has both “works_for_me” and “passes_guidelines” tags, we will promote the module for inclusion in Ansible Extras.  At this point, you will be expected to maintain the module by fixing bugs and evaluating pull requests in a timely manner. 

Thanks again for submitting your Ansible module!

**Comment by gregdek** [comment]
Thanks @bcoca for this PR. This PR requires revisions, either because it fails to build or by reviewer request. Please make the suggested revisions. When you are done, please comment with text 'ready_for_review' and we will put this PR back into review.

[This message brought to you by your friendly Ansibull-bot.]

**Comment by gregdek** [comment]
@bcoca A friendly reminder: this pull request has been marked as needing your action. If you still believe that this PR applies, and you intend to address the issues with this PR, just let us know in the PR itself and we will keep it open pending your changes. When you do address the issues, please respond with ready_for_review in your comment, so that we can notify the maintainer.

[This message brought to you by your friendly Ansibull-bot.]

**Comment by gregdek** [comment]
@bcoca Another friendly reminder: this pull request has been marked as needing your action. If you still believe that this PR applies, and you intend to address the issues with this PR, just let us know in the PR itself and we will keep it open. If you have addressed the issues and believe it's ready for review, please comment with the text "ready_for_review". If we don't hear from you within another 14 days, we will close this pull request.

[This message brought to you by your friendly Ansibull-bot.]

**Comment by gregdek** [comment]
@bcoca A friendly reminder: this pull request has been marked as needing your action. If you still believe that this PR applies, and you intend to address the issues with this PR, just let us know in the PR itself and we will keep it open pending your changes. When you do address the issues, please respond with ready_for_review in your comment, so that we can notify the maintainer.

[This message brought to you by your friendly Ansibull-bot.]

**Comment by gregdek** [comment]
Thanks @bcoca for this PR. This PR requires revisions, either because it fails to build or by reviewer request. Please make the suggested revisions. When you are done, please comment with text 'ready_for_review' and we will put this PR back into review.

[This message brought to you by your friendly Ansibull-bot.]

**Comment by gregdek** [comment]
@bcoca A friendly reminder: this pull request has been marked as needing your action. If you still believe that this PR applies, and you intend to address the issues with this PR, just let us know in the PR itself and we will keep it open pending your changes. When you do address the issues, please respond with ready_for_review in your comment, so that we can notify the maintainer.

[This message brought to you by your friendly Ansibull-bot.]

**Comment by gregdek** [comment]
@bcoca Another friendly reminder: this pull request has been marked as needing your action. If you still believe that this PR applies, and you intend to address the issues with this PR, just let us know in the PR itself and we will keep it open. If you have addressed the issues and believe it's ready for review, please comment with the text "ready_for_review". If we don't hear from you within another 14 days, we will close this pull request.

[This message brought to you by your friendly Ansibull-bot.]

File: openweather_facts.py [hunk 1]
```diff
@@ -0,0 +1,173 @@
...
+DOCUMENTATION = '''
+---
+module: openweather_facts
+author:
+    - '"Brian Coca (@bcoca)" <bcoca@ansible.com>'
+version_added: "2.0"
+description:
+    - Use the openweathermap API to retrieve the current weather at a location.
+    - Even though none of the fields are required, you must supply one (and only one) for the query.
+    - You can get more info on the return data and codes at http://openweathermap.org/current
```
Comment (gundalow) [inline openweather_facts.py]: `U(http://openweathermap.org/current)`

File: openweather_facts.py [hunk 2]
```diff
@@ -0,0 +1,173 @@
...
+    required: false
+    default: null
+  units:
+    description:
+      - What units you want in the response, the default null is the same as specifying 'internal'
+    required: False
+    default: null
+    choices: ['internal', 'metric', 'imperial']
+note:
+  - This module uses the free tier of openweathermap.org, it should be easy to expand to use a developer or higher tier
```
Comment (gundalow) [inline openweather_facts.py]: `U(http://openweathermap.org)`

File: openweather_facts.py [hunk 3]
```diff
@@ -0,0 +1,173 @@
...
+
+    data = json.loads(response.read())
+    weather['weather_' + str(data['id'])] = data
+
+    module.exit_json(info=info, ansible_facts=weather, uri=uri)
+
+# import module snippets
+from ansible.module_utils.basic import *
+from ansible.module_utils.urls import *
+main()
```
Comment (gundalow) [inline openweather_facts.py]: `main()` needs to be in `if`

**Comment by gundalow** [comment]
Hi,

Looks like this branch needs updating to fix Travis Issues, please see this guide on how to do that cleanly.

http://docs.ansible.com/ansible/community.html#contributing-code-features-or-bugfixes

needs_revision

**Comment by gregdek** [comment]
Thanks @bcoca for this PR. This PR requires revisions, either because it fails to build or by reviewer request. Please make the suggested revisions. When you are done, please comment with text 'ready_for_review' and we will put this PR back into review.

[This message brought to you by your friendly Ansibull-bot.]

**Comment by gregdek** [comment]
@bcoca A friendly reminder: this pull request has been marked as needing your action. If you still believe that this PR applies, and you intend to address the issues with this PR, just let us know in the PR itself and we will keep it open pending your changes. When you do address the issues, please respond with ready_for_review in your comment, so that we can notify the maintainer.

[This message brought to you by your friendly Ansibull-bot.]

**Comment by gregdek** [comment]
@bcoca Another friendly reminder: this pull request has been marked as needing your action. If you still believe that this PR applies, and you intend to address the issues with this PR, just let us know in the PR itself and we will keep it open. If you have addressed the issues and believe it's ready for review, please comment with the text "ready_for_review". If we don't hear from you within another 14 days, we will close this pull request.

[This message brought to you by your friendly Ansibull-bot.]

**Comment by gregdek** [comment]
Thanks @bcoca for this PR. This PR requires revisions, either because it fails to build or by reviewer request. Please make the suggested revisions. When you are done, please comment with text 'ready_for_review' and we will put this PR back into review.

[This message brought to you by your friendly Ansibull-bot.]

**Comment by bcoca** [comment]
closing in favor of https://github.com/ansible/ansible/pull/20880

**Pull request closed without merging**